src/datasets/builder.py

```diff
@@ -2081,3 +2078,56 @@ def _build_pcollection(pipeline):
 
         # Add the PCollection to the pipeline
         _ = pipeline | split_name >> _build_pcollection()  # pylint: disable=no-value-for-parameter max_bytes_per_shard
+
+    def as_streaming_dataset(
+        self,
+        split: Optional[str] = None,
+    ) -> Union[Dict[str, IterableDataset], IterableDataset]:
+        self._request_info_from_hf_gcs()
```

Comment:
Maybe raise NotImplementedError if the dataset is not on the HF GCS ?

Reply:
This method already may raise `DatasetNotOnHfGcsError`. You don't like this error and you prefer `NotImplementedError` instead?

Reply:
Ah yes indeed, it's fine then I think